rfcs/1158-type-stability.md

```diff
@@ -280,41 +371,128 @@ In practice, this suggests two key considerations around type bugs:
 
-These type tests should be specific and precise. It is important, for example, to guarantee that an API element never *accidentally* becomes `any`, thereby making many things allowable which should not be in the case of function arguments, and "infecting" the caller's code by eliminating type safety on the result in the case of function return values. The `expect-type` library's `.toEqualTypeOf` assertion is robust against precisely this scenario; addon authors are also encouraged to use its `.not` modifier and `.toBeAny()` method where appropriate to prevent this failure mode.
+- [`tsd`][tsd] -- a full solution for testing types by writing `.test-d.ts` files with a small family of assertions and using the `tsd` command to validate all `.test-d.ts` files. Authoring has robust editor integration, since the type assertions are normal TS imports, and the type assertions are specific enough to catch all the kinds of breakage identified above. It is implemented using the TS compiler version directly, which makes its assertions fairly robust. Risks and downsides:
 
-This has no impact on runtime code: the `types` directory is not used for generating builds. However, to be safe, these tests should *never* export any code. They should import the public API they are testing and `expectTypeOf` and any test helpers, and nothing else. Actually executing the test then consists of running `tsc --noEmit`.
+    -   The tool uses a patched version of the TypeScript compiler, which increases the risk of errors and the risk that at some points it will simply be unable to support a new version of TypeScript.
 
+    -   Because the assertions are implemented as type definitions, the library is subject to the same risk of compiler breakage as the types it is testing.
+
+    -   **BLOCKER:**  currently only supports a single version of TypeScript at a time. While the author is [interested][tsd-versions] in supporting multiple versions, it is not currently possible.
+
+- [`expect-type`][expect-type] -- a library with a variety of type assertions, inspired by Jest's matchers but tailored to types and with no runtime implementation. Like `tsd`, it is implemented as a series of function types which can be imported, and accordingly it has excellent editor integration. However, unlike `tsd`, it does *not* use the compiler API. Instead,  It is robust enough to catch all the varieties of breaking type changes. The risks with expect-type are:
+
+    -   It currently has a single maintainer, and relatively few users.
+
+    -   It is relatively young, having been created only about a year ago, and therefore having existed for only 5 TypeScript releases. While its track record is good so far, there is not yet evidence of how it would deal with serious breaking changes like those introduce in TypeScript 3.5.
+
+    -   BBecause the assertions are implemented as type definitions, the library is subject to the same risk of compiler breakage as the types it is testing.
```

Comment:
```suggestion
    -   Because the assertions are implemented as type definitions, the library is subject to the same risk of compiler breakage as the types it is testing.
```

Reply:
Ahh, yes! Thank you. I fixed this one in 3f51dfe!